File: backend/app/schemas/nearest_stop.py
from pydantic import BaseModel


class UserLocationResponse(BaseModel):
    lat: float
    lon: float


class NearestStopResponse(BaseModel):
    stop_id: str
    name: str
    street_name: str | None = None
    lat: float
    lon: float
    sequence: int | None = None
    distance_meters: float


class NearestStopsResponse(BaseModel):
    pattern_id: str
    user_location: UserLocationResponse
    nearest_stops: list[NearestStopResponse]

File: frontend/src/types/nearestStop.ts
export interface NearestStop {
  stop_id: string;
  name: string;
  street_name?: string | null;
  lat: number;
  lon: number;
  sequence?: number | null;
  distance_meters: number;
}

export interface NearestStopsResponse {
  pattern_id: string;
  user_location: {
    lat: number;
    lon: number;
  };
  nearest_stops: NearestStop[];
}
